code/_onclick/item_attack.dm

```diff
@@ -26,12 +26,14 @@
 /obj/item/proc/pre_attack(atom/A, mob/living/user, params) //do stuff before attackby!
 	if(SEND_SIGNAL(src, COMSIG_ITEM_PRE_ATTACK, A, user, params) & COMPONENT_CANCEL_ATTACK_CHAIN)
 		return TRUE
-	if(is_hot(src) && A.reagents && !ismob(A))
+
+	var/temperature = get_heat()
+	if(temperature && A.reagents && !ismob(A) && !istype(A, /obj/item/clothing/mask/cigarette))
```

Comment:
I really dislike this istype check. Maybe you can make it part of the signal COMSIG_ITEM_PRE_ATTACK?